doc: quickstart for Go  projects (#3551)
operator-framework/operator-sdk: website/content/en/docs/building-operators/golang/references/env-test-setup.md

> 

Reviewer: Nits:
1. Change the filename to `envtest-setup.md`
2. Change the title and linkTitle to `Envtest Setup`
3. Add description: `Learn how to setup your project to run integration tests using envtest`

> ## Installing prerequisites

Reviewer: I don't think we need the `Overview` and `Installing prerequisites` headers in this file. Right after the frontmatter, I would go right into this:

> By default, Go-based operators are scaffolded to make use of controller-runtime's [`envtest`][envtest] framework, which uses `kubectl`, `kube-apiserver`, and `etcd` to simulate the API portions of a real cluster. You can use [this script][script] to download ...

> [Envtest][envtest] requires that `kubectl`, `api-server` and `etcd` be present locally. You can use this [script][script] to download these binaries into the `testbin/` directory which will be created in your project. Update your Makefile by replacing your `test` target with: 

Reviewer: ```suggestion
By default, Go-based operators are scaffolded to make use of controller-runtime's [`envtest`][envtest] framework, which uses `kubectl`, `kube-apiserver`, and `etcd` to simulate the API portions of a real cluster. You can use [this script][script] to download these binaries into the `testbin/` directory and configure your environment to use them. Update your `Makefile` by replacing your `test` target with: 
```

```sh
# Run tests
ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
test: generate fmt vet manifests
        mkdir -p ${ENVTEST_ASSETS_DIR}
        test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/master/hack/setup-envtest.sh
        source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); go test ./... -coverprofile cover.out
```

> Also, it is recommended add into the `.gitignore` a new line with `testbin/*` for you do not commit these binaries. 

Reviewer: ```suggestion
If using `git`, it is recommended to add `testbin/*` to your `.gitignore` file to avoid committing these binaries. 
```

> See that you can also use your own binaries and change the location via setting up the following environment variables in your `controllers/suite_test.go`: 

Reviewer: I think I would suggest removing this and everything below.

The above description is short, straightforward, descriptive, and can get users up and running. I think that's all we need.
